cmd/livepeer/starter/starter.go

```diff
@@ -1227,10 +1227,27 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 					}
 				}
 
-				// If the config contains a URL we call Warm() anyway because AIWorker will just register
-				// the endpoint for an external container
 				if config.Warm || config.URL != "" {
+					// Register the external endpoint if URL is provided
 					endpoint := worker.RunnerEndpoint{URL: config.URL, Token: config.Token}
+
+					// If config.Warm is true, setup the AI worker with the selected Docker image
+					if config.Warm && config.URL == "" {
+						// Determine which image to use based on the pipeline
+						imageToUse := *cfg.AIRunnerImage // Default to the image from config
+						if specificImage, ok := core.PipelineToImage[config.Pipeline]; ok {
+							imageToUse = specificImage
+						}
+
+						// Setup the AI worker with the selected Docker image
+						n.AIWorker, err = worker.NewWorker(imageToUse, gpus, modelsDir)
```

Comment:
@pschroedll Unfortunately, this logic won't work when starting multiple containers. Here's what I suggest we do instead:

- Rename the [containerImageID](https://github.com/pschroedl/ai-worker/blob/50fcc14304dc4b6d7ea8a7cf055b32a35d9d5768/worker/docker.go#L43) to `containerImageIDDefault`.
- Add an optional `containerImageID` parameter to the [function](https://vscode.dev/github/pschroedl/ai-worker/blob/segment_anything_2_pipeline_image/worker/docker.go#L157) and set it in the container config when present.
- Apply the same logic to [docker.go/DockerManager/Borrow](https://github.com/pschroedl/ai-worker/blob/50fcc14304dc4b6d7ea8a7cf055b32a35d9d5768/worker/docker.go#L110).
- Apply the same logic to [docker.go/DockerManager/Warm](https://github.com/pschroedl/ai-worker/blob/50fcc14304dc4b6d7ea8a7cf055b32a35d9d5768/worker/docker.go#L79).
- This will allow us to simplify the logic in the starter as well.

Let me know your thoughts!

Reply:
Sounds like a good approach to me - thanks @rickstaa !

Reply:
After testing this approach, I realized that a better approach might be to delegate custom image mapping down to the AI-Worker. 

One motivation is that Borrow is also called directly from ai-worker itself, leading to a state where the the custom containerImageID name:tag was unavailable.